ocaml/testsuite/tests/typing-layouts-err-msg/error_messages.ml

```diff
@@ -76,28 +76,26 @@ Line 3, characters 21-22:
 Error: This expression has type v but an expression was expected of type
          'a t = ('a : void)
        The layout of v is immediate, because
-         it's an enumeration variant (all constructors are constant).
+         of the definition of v at line 2, characters 0-10.
        But the layout of v must be a sublayout of void, because
-         of the annotation on 'a in the declaration of the type t.
+         of the definition of t at line 1, characters 0-22.
 |}]
 
 (* Extensible_variant *)
 type ('a: void) t = 'a
 type attr = ..
-let f (x: attr): 'a t = x
+and w = attr t
```

Comment:
Why change this test?

Reply:
Motivation is to keep the `it's an extensible variant` error message.

With the original test, the layout reason gets hidden:
```
Error: This expression has type attr but an expression was expected of type
         'a t = ('a : void)
       The layout of attr is value, because
         of the definition of attr at file "./debug_test.ml", line 246, characters 0-14.
       But the layout of attr must be a sublayout of void, because
         of the definition of t at file "./debug_test.ml", line 245, characters 0-22.
```